uber/menu.py

```diff
@@ -0,0 +1,95 @@
+from uber.common import *
+
+
+class MenuItem:
+    access = None   # list of permission levels allowed to display this menu
+    href = None     # link to render
+    submenu = None  # submenu to show
+    name = None     # name of Menu item to show
+
+    def __init__(self, href=None, access=None, submenu=None, name=None):
+        assert submenu or href, "menu items must contain ONE nonempty: href or submenu"
+        assert not submenu or not href, "menu items must not contain both a href and submenu"
+
+        if submenu:
+            self.submenu = listify(submenu)
+        else:
+            self.href = href
+
+        self.name = name
+        self.access = access
+
+    def append_menu_item(self, m):
+        """
+        If we're appending a new menu item, and we aren't a submenu, convert us to one now.
+        Create a new submenu and append a new item to it with the same name and href as us.
+
+        Example:
+            Original menu:
+                (name='Rectangles', href="rectangle.html')
+
+            Append to it a new menu item:
+                [name='Squares', href='square.html']
+
+            New result is:
+                (name='Rectangles', submenu=
+                    [
+                        (Name='Rectangles', href="rectangle.html"),
+                        (Name='Squares', href="square.html")
+                    ]
+                )
+        """
+        if not self.submenu and self.href:
+            self.submenu = [MenuItem(name=self.name, href=self.href)]
+            self.href = None
+
+        self.submenu.append(m)
+
+    def render_items_filtered_by_current_access(self):
+        """
+        Returns: dict of menu items which are allowed to be seen by the logged in user's access levels
+        """
+        out = {}
+
+        allowed = True
+        if self.access:
```

Comment:
I think this block can be optimized. Off the top of my head:
```python
# Immediately return None if the admin does not have any of the access levels listed
if self.access and set(self.access).isdisjoint(sa.AdminAccount.access_set()):
  return None
```

Something that would be nice to do in the future is also include a way to require multiple access levels at once before a link is displayed. That's out of scope for this PR though, I think.